src/System.Text.Formatting/System/Text/Formatting/ITextOutputExtensions.cs

```diff
@@ -303,18 +303,23 @@ public static class ITextOutputExtensions
         public static void Append<TFormatter>(this TFormatter formatter, double value, StandardFormat format = default) where TFormatter : ITextOutput
         {
             while (!formatter.TryAppend(value, format)) {
-                formatter.Enlarge();
+                formatter.GetMoreMemory();
             }
         }
 
         public static bool TryAppend<TFormatter>(this TFormatter formatter, double value, StandardFormat format = default) where TFormatter : ITextOutput
         {
-            if (!CustomFormatter.TryFormat(value, formatter.GetSpan(), out int bytesWritten, format, formatter.SymbolTable))
+            if (!CustomFormatter.TryFormat(value, formatter.GetMemory().Span, out int bytesWritten, format, formatter.SymbolTable))
             {
                 return false;
             }
             formatter.Advance(bytesWritten);
             return true;
         }
+
+        private static void GetMoreMemory<TFormatter>(this TFormatter formatter) where TFormatter : ITextOutput
+        {
+            formatter.GetMemory(formatter.GetMemory().Length * 2);
```

Comment:
Yeah, this looks inefficient.

Reply:
These API's have to be on OuputWriter anyway.